### ISSUE TITLE
Allow for live refresh when tracks are added via Web API

### ISSUE THREAD

**Issue by crwgregory** [spotify/web-api#218]
TLDR:

- Spotify Desktop Client needs to be made aware of changes to a playlist when made by the Rest Api. 
- Or support for 'Add to Queue' needs to be added.

Hello everyone,

I've been hard at work over the past 2 months developing an application that I am extremely excited about and have been spending all of my free time working on. Staying for hours at the office after the normal work day has ended and everyone's gone home, also spending entire weekends on it. I really believe in this app and want nothing more than to see it come to life. I've never put so much time and energy into an idea!

I've learned how to developed using Node, Gulp, Babel, Sockets, SQLite, Webpack, Javascript Modules, Javascript Promises, Javascript Classes(!). I've learned so much! Not to mention the Spotify Web API. I was not expecting such a developed and user friendly API when I first started this adventure!

This weekend I was super excited because I thought I would have a working product to bring to the office Monday and show it off to my coworkers and test it out, however, it was brought to a  heart breaking halt. I found out that the Spotify Desktop Client does not refresh playlists when tracks are added, re-ordered, or deleted through the web API (a hard lesson in checking the actuality of an idea before implementing).

But I didn't give up! I searched high and low for a solution, Spotify Web Widgets, IFrame manipulation, automatic play, embedding YouTube videos, nothing was working. There's been a couple 'new idea' requests on the Spotify community page [here](https://community.spotify.com/t5/Closed-Ideas/quot-Refresh-playlists-quot-button-menu-option/idi-p/570392) and [here](https://community.spotify.com/t5/Spotify-Ideas/Refresh-all-portions-of-Spotify-with-settings-feature/idi-p/1042517) about this feature, but while being good ideas, never gained enough traction in the community. So I am here, submitting it again but to be implemented in the Web API.

For web interactions to be as user friendly as possible, updates should be seen automatically when the user makes hes/her choice. Adding this functionality to the API would allow for developers to integrate Spotify into new and exciting applications that can update a user's playlist while the user is listening to the playlist. I cannot even imagine some of the applications that could take advantage of this feature, but some might be:

<section>
<ul>
<li>Voice control to add moods/seeds to a playlist live time</li>
<li>Changes in playlist when user's heart rate increases (could be useful for exercise apps).</li>
</ul>
</section>

And many, many more uses!

I don't know how exactly it would work but I imagine a server sent event would be dispatched to the desktop client. I don't know the specifics of how Spotify talks to the client, but Spotify Web Widgets already seem to have this capability.

I would be eternally grateful if the Spotfiy team could make this request a reality. I've put so much blood, sweet, and tears into this app to see it stopped. I am at your mercy, Oh Great Spotfiy!

<section>
<h5>Requirements: Update/Refresh playlist (playlist_id) when tracks are:</h5>
<ul>
<li>Added</li>
<li>Re-ordered</li>
<li>Deleted</li>
</ul>
</section>

**Comment by AdrielGiannoni** [comment]
Hello @crwgregory, i have a similar problem like yours.
I am working with WEB API. I can add, reorder or delete tracks in a playlist without problem using a POST method, and then windows client spotify refresh instantly. But later, some hours after my service was working adding some tracks to the playlist, the client STOP refreshing, and i couldnt see the tracks that were added. So i have to restart the spotify client to continue working properly. This is similar for android and windows app. 
if someone has some help i will be grateful. i am working with .net c#
Sorry about my english.

**Comment by asmitter** [comment]
As you describe, the issue is that the desktop client can be slow to update after tracks are added via the Web API. What you experience is related to the desktop client rather than the APIs, so I'll close this.

However, I did just try this out today and was happy to see tracks appearing immediately for some time.

**Comment by crwgregory** [comment]
Is that right? Did something change on the Desktop Client?

**Comment by AdrielGiannoni** [comment]
I was doing a test.
My spotify client STOP sync (refreshing tracks that was added by API) when my PC loss the network connection for one or two seconds. So then, spotify client dont refresh any more, and i have to reset it.
Definitely, I think that Spotify client stop listening events of refreshing by API when i have a micro networks interruption for some miliseconds, and never more reconnect to events. So, i have to restart client.

Note: I made this test with two diferents PC, and the last version of the client.
Thanks.

**Comment by asmitter** [comment]
Again, this will be due to the desktop client rather than any issues with the APIs, and this tracker is for issues with the latter. I see a post in the community which looks like it might be yours: https://community.spotify.com/t5/Live-Ideas/All-Platforms-Playlists-Allow-for-live-playlist-updates-Auto/idi-p/1328578 ?
I recommend you follow up there so that the right people can be looped in.

**Comment by crwgregory** [comment]
Yup that's mine. 5 kudos so far. Ridding that kudos train! 

Honestly though it doesn't look to be gaining much traction.

This is not just a Desktop Client issue/new idea. Unless the webplayer that makes requests to the Desktop Client ( open.spotify.com ) is considered part of the Desktop Client. 

But you're right, this is an issue tracker for the web api. This is not an issue, more of a feature request. Would the feature of adding songs to the spotify queue _using_ the web api not be a web api issue? Just as the web api is used to add tracks to a playlist. 

Would you suggest that I make a new feature request of just 'Add tracks to queue' ?

**Comment by universallythinking** [comment]
I solved this problem by making batch files, and vbs files to delete Spotify's cache every hour or so...  Can be done in AppleScript as well.

**Comment by zearg** [comment]
Here is my batch script : https://gitlab.com/snippets/1698793
(released in few minutes, without any knowledge in batch language ^^)
